data-prepper-plugins/kafka-plugins/src/main/java/org/opensearch/dataprepper/plugins/kafka/source/KafkaSource.java

```diff
@@ -24,15 +42,96 @@
+          executorService.submit(multithreadedConsumer);
+        });
+      } catch (Exception e) {
+        LOG.error("Failed to setup the Kafka Source Plugin.", e);
+        kafkaWorkerThreadProcessingErrors.increment();
+      }
+    });
   }
 
   @Override
   public void stop() {
-   LOG.info("Initiated the stop method...");
+    LOG.info("Shutting down Consumers...");
+    executorService.shutdown();
+    try {
+      if (!executorService.awaitTermination(
+              sourceConfig.getTopics().get(0).getTopic().getConsumerGroupConfig().getThreadWaitingTime().toSeconds(), TimeUnit.MILLISECONDS)) {
+        LOG.info("Consumer threads are waiting for shutting down...");
+        executorService.shutdownNow();
+      }
+    } catch (InterruptedException e) {
+      if (e.getCause() instanceof InterruptedException) {
+        LOG.error("Interrupted during consumer shutdown, exiting uncleanly...");
+        executorService.shutdownNow();
+        Thread.currentThread().interrupt();
+      }
+    }
+    LOG.info("Consumer shutdown successfully...");
+  }
+
+  private Properties getConsumerProperties(TopicsConfig topicConfig) {
+    Properties properties = new Properties();
+    try {
+      properties.put(ConsumerConfig.AUTO_COMMIT_INTERVAL_MS_CONFIG,
+              topicConfig.getTopic().getConsumerGroupConfig().getAutoCommitInterval().toSecondsPart());
+      properties.put(ConsumerConfig.AUTO_OFFSET_RESET_CONFIG,
+              topicConfig.getTopic().getConsumerGroupConfig().getAutoOffsetReset());
+      properties.put(ConsumerConfig.BOOTSTRAP_SERVERS_CONFIG, sourceConfig.getBootStrapServers());
+      properties.put(ConsumerConfig.ENABLE_AUTO_COMMIT_CONFIG,
+              topicConfig.getTopic().getConsumerGroupConfig().getAutoCommit());
+      properties.put(ConsumerConfig.GROUP_ID_CONFIG, topicConfig.getTopic().getConsumerGroupConfig().getGroupId());
+      setPropertiesForSchemaType(topicConfig, properties, topicConfig.getTopic().getSchemaConfig().getSchemaType());
+    } catch (Exception e) {
+      LOG.error("Unable to create the Kafka Consumer from the given configurations.", e);
+    }
+
+    return properties;
   }
 
+  private void setPropertiesForSchemaType(TopicsConfig topicConfig, Properties properties, String schemaType) {
+
+    if (schemaType.equalsIgnoreCase(MessageFormat.JSON.toString())) {
+      properties.put(ConsumerConfig.KEY_DESERIALIZER_CLASS_CONFIG,
+              topicConfig.getTopic().getSchemaConfig().getKeyDeserializer());
+      properties.put(ConsumerConfig.VALUE_DESERIALIZER_CLASS_CONFIG, JsonDeserializer.class);
+    } else if (schemaType.equalsIgnoreCase(MessageFormat.PLAINTEXT.toString())) {
+      properties.put(ConsumerConfig.KEY_DESERIALIZER_CLASS_CONFIG,
+              topicConfig.getTopic().getSchemaConfig().getKeyDeserializer());
+      properties.put(ConsumerConfig.VALUE_DESERIALIZER_CLASS_CONFIG,
+              topicConfig.getTopic().getSchemaConfig().getValueDeserializer());
+    }
```

Comment:
else? if it's not Json/Plaintext should we throw an exception? also would be good to add a log message on schema type

Reply:
Log message added.
No need of throwing exception as the default type is JSON.